### PR TITLE
Pass **kwargs to solvers

### PR DESCRIPTION
**Pull request by kinnala**
@gdmcbain : is this what you propose in #300 ?

### CODE REVIEW

File: skfem/utils.py [hunk 1]
```diff
@@ -204,14 +191,15 @@ def solver(A, b):
 
 def solver_iter_pcg(**kwargs) -> LinearSolver:
     """Conjugate gradient solver, specialized from solver_iter_krylov"""
-    return solver_iter_krylov(spl.cg, **kwargs)
+    return solver_iter_krylov(**kwargs)
 
 
 def solve(A: spmatrix,
```
Comment (kinnala) [inline skfem/utils.py]: See, in particular, this part and the related changes in all of the solvers.

**Comment by gdmcbain** [comment]
No, I didn't foresee not being able to define a solver with a fixed preconditioner.

File: skfem/utils.py [hunk 2]
```diff
@@ -160,8 +153,7 @@ def build_pc_diag(A: spmatrix) -> spmatrix:
 
 
 def solver_iter_krylov(krylov: Optional[LinearSolver] = spl.cg,
-                       verbose: Optional[bool] = False,
-                       **kwargs) -> LinearSolver:
```
Comment (gdmcbain) [inline skfem/utils.py]: I think we do still want to be able to parameterize Krylov solvers on definition; see #307.

File: skfem/utils.py [hunk 3]
```diff
@@ -187,7 +174,7 @@ def callback(x):
         if verbose:
             print(np.linalg.norm(x))
 
-    def solver(A, b):
+    def solver(A, b, **kwargs):
```
Comment (gdmcbain) [inline skfem/utils.py]: Yes, we do want to accept here keyword-arguments for passing on to the solver at solve-time but taking care not to smother those passed in here for defining the solver.  I'm thinking of cases like a heat equation for which the solver is used repeatedly with the same operator `A`, preconditioner `M`, &c.,  but changing right-hand side `b` and starting guess `x0`.

Comment (gdmcbain) [reply skfem/utils.py]: See #307.

Comment (kinnala) [reply skfem/utils.py]: Makes sense, let's go with this approach.

**Comment by kinnala** [comment]
Sorry for doing the reordering in the same PR. I didn't realize the changelog would become so unreadable. I'll try to avoid this in the future.

**Pull request merged**